hw/top_earlgrey/dv/chip_sim_cfg.hjson

```diff
@@ -186,13 +186,27 @@
               --out {run_dir}/otp_ctrl_img_raw.vmem \
               {gen_otp_images_cmd_opts}
         ''',
-         '''{gen_otp_images_cmd} \
+        '''{gen_otp_images_cmd} \
               --img-cfg {gen_otp_images_cfg_dir}/otp_ctrl_img_dev.hjson \
```

Comment:
It looks to me like the `otp_ctrl_img_[dev|prod|rma].hjson` files specify values for distinct partitions from the `otp_ctrl_img_[creator_sw|owner_sw|hw]_cfg.hjson` files, so I don't think it makes a difference, but it seems more logical to me to apply the lifecycle specific overlay last rather than first.

Reply:
Agreed that it seems more intuitive to end with the lifecycle-specific overlay. However, I ran into this issue in #15407 which is that the `gen-otp-img.py` script, when constructing the `OtpMemImg` object, checks for the presence of a `seed` field, which is only specified in the lifecycle-specific hjsons. It doesn't perform this check when applying overlays though, so I imagine that's why it's done "backwards" here.

See the code for [constructing an OtpMemImg](https://cs.opensource.google/opentitan/opentitan/+/master:util/design/lib/OtpMemImg.py;l=150) vs the code for [applying overlays](https://cs.opensource.google/opentitan/opentitan/+/master:util/design/lib/OtpMemImg.py;l=325).